frontend/public/components/masthead.tsx

```diff
@@ -23,6 +23,10 @@ const UserMenu: React.StatelessComponent<UserMenuProps> = ({username, actions})
     <i className="fa fa-user co-masthead__user-icon" aria-hidden="true"></i>
     <span className="co-masthead__username">{username}</span>
   </React.Fragment>;
+  if (_.isEmpty(actions)) {
```

Comment:
Not a block, just a nit, but I'd suggest:

```JavaScript 
return _.isEmpty(actions) 
  ? title 
  : <ActionsMenu actions={actions}
       title={title}
       noButton={true} />;
```
A little style, but more keeping the idea of avoiding side effects front and center.

Reply:
(and this is a really easy one to keep consistent)

Reply:
I think this is just a place we disagree on style. I find the early return easier to read.

Reply:
> Avoiding side-effects

I don't understand what you mean by side effect here. Side effect usually refer to data being changed like an argument when a function is called. I don't see a side effect.

> Second, 1 entry, 1 exit (return once). Also trivial here, but helps when things get more complicated (encourages breaking functions down into smaller bits).

Early returns are a terrific pattern. They are very easy to reason about, particularly when there is more logic below the return (not in this case, but in general). I dislike making code harder to read in order to always follow this rule. We used early returns effectively in origin-web-console.

So again I just don't agree :/

Reply:
Right!  `ternary` + single `return` = avoid `more logic below the return`, make smaller functions :) 

But I also see that `golang` loves lots of returns, so woohoo!

/lgtm as well though, just to clarify this is more of a kicks-and-giggles discussion.